smif/data_layer/__init__.py

```diff
@@ -1,3 +1,398 @@
+
+    @abstractmethod
+    def read_interval_set_data(self, interval_set_name):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_units(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_region_set(self, data):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_interval_set(self, data):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_units(self, data):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_scenario_sets(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_scenarios(self, scenario_set_name):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_scenario_data(self, scenario_name):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_scenario_set(self, scenario_set):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_scenario(self, scenario):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_scenario_data(self, scenario_name, data):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_narrative_sets(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_narratives(self, narrative_set_name):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_narrative_data(self, narrative_name):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_narrative_set(self, narrative_set):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_narrative(self, narrative):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_narrative_data(self, narrative_set_name, data):
+        raise NotImplementedError()
+
+
+class DatafileInterface(DataInterface):
+    """ Read and write interface to YAML / CSV configuration files
+    """
+    def __init__(self, config_path):
+        """Initialize file paths
+        """
+        config_path = str(config_path)
+        self.filepath = {
+            'sos_model_runs': os.path.join(config_path, 'sos_model_runs'),
+            'sos_models': os.path.join(config_path, 'sos_models'),
+            'sector_models': os.path.join(config_path, 'sector_models')
+        }
+
+    def read_sos_model_runs(self):
+        """Returns a list of sos_model_run dictionaries
+        """
+        sos_model_runs = list()
+
+        sos_model_run_names = self._read_yaml_filenames_in_dir(self.filepath['sos_model_runs'])
+        for sos_model_run_name in sos_model_run_names:
+            sos_model_runs.append(self._read_yaml_file(self.filepath['sos_model_runs'],
+                                                       sos_model_run_name))
+
+        return sos_model_runs
+
+    def write_sos_model_run(self, sos_model_run):
+        """Write sos_model_run dictionary to Yaml file
+        Existing configuration will be overwritten without warning
```

Comment:
Leave a space after the first line of the docstring - the first line should summarise the method, any lines below that (after the empty line) give more detail.